pkg/framework/test/etcd_config_test.go

```diff
@@ -1,36 +1,68 @@
-package test_test
+package test
 
 import (
-	. "k8s.io/kubectl/pkg/framework/test"
+	"fmt"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("EtcdConfig", func() {
-	It("does not error on valid config", func() {
-		conf := &EtcdConfig{
-			PeerURL:   "http://this.is.some.url:1234",
-			ClientURL: "http://this.is.another.url/",
-		}
-		err := conf.Validate()
-		Expect(err).NotTo(HaveOccurred())
-	})
+	Context("Constructor", func() {
+		var (
+			previousPortFinder PortFinder
+		)
+		BeforeEach(func() {
+			previousPortFinder = etcdPortFinder
+		})
+		AfterEach(func() {
+			etcdPortFinder = previousPortFinder
+		})
+
+		It("sets some sane default URLs", func() {
+			etcdPortFinder = func(host string) (port int, addr string, err error) {
```

Comment:
Yeah, that's a little unfortunate. I see that as a sign that we're doing something wrong. Do you know what it is?

Reply:
We might want to pass in the portfinder, with it being optional? Might be related to my earlier comment.